### PR TITLE
Deletion of added todo-item feature

### PR DESCRIPTION
**Pull request by coder4097**
### Endpoint to delete added todo item 

URI: DELETE /todos/:id
res.status: 204 when provided valid `id`

### Database schema changes
N/A

### Automated test cases added
- Return 204 if `id` provided is valid
- return 404 if `id` not found

### Manual test cases run
Done in postman

### CODE REVIEW

**Comment by coder1033** [comment]
Please recheck your manual test description.

File: src/controllers/todo-item-controller.ts [hunk 1]
```diff
@@ -8,7 +8,8 @@ import {
   ExtendedRequest,
   ValidationFailure,
 } from '@typings';
-import { createTodoItemValidator } from '@validators';
+import { createTodoItemValidator, deleteTodoItemValidator } from '@validators';
+import { ValidationError } from 'src/typings/error';
```
Comment (coder1033) [inline src/controllers/todo-item-controller.ts:12]: Are you using `ValidationError` in this file?

File: src/controllers/todo-item-controller.ts [hunk 2]
```diff
@@ -50,4 +57,29 @@ export class TodoItemController extends BaseController {
...
+      return next(valError);
+    }
+    const { id } = req.params;
+    const todoItem = await this.appContext.todoItemRepository.deleteMany({
+      _id: id,
+    });
+    if (todoItem.deletedCount > 0) {
+      res.status(204).json(todoItem.deletedCount);
+    } else {
+      res.status(404).send();
```
Comment (coder1033) [inline src/controllers/todo-item-controller.ts:82]: 404 is already implemented in boilerplate. Use `Errors.NotFoundError` here.

**Comment by coder1033** [comment]
We don't delete any data from database.

Add a boolean `isActive` in `todoItemSchema`, which is by default `true`.
If `isActive` is `true`, then data is available for client.
If `isActive` is `false`, then data is not available for client.

File: src/controllers/todo-item-controller.ts (hunk 1 above)
Comment (coder4097) [reply src/controllers/todo-item-controller.ts:12]: yes..line 45

File: src/controllers/todo-item-controller.ts (hunk 2 above)
Comment (coder4097) [reply src/controllers/todo-item-controller.ts:82]: okay
updated

File: test/spec/todo/todo-item.spec.ts [hunk 3]
```diff
@@ -58,3 +58,28 @@ describe('POST /todos', () => {
       .to.equal('Please specify the valid title');
   });
 });
+
+describe('DELETE /todos/:id', () => {
+  it('should return 204 if valid mongoId is passed and todo item exists', async () => {
+    let todoItem = await testAppContext.todoItemRepository.save(
```
Comment (coder1033) [inline test/spec/todo/todo-item.spec.ts:64]: `const`

**Review by coder1033 — APPROVED** [review]

**Pull request merged**